plugins/PagePerformance/Tracker/PerformanceDataProcessor.php

```diff
@@ -0,0 +1,85 @@
+<?php
+/**
+ * Matomo - free/libre analytics platform
+ *
+ * @link https://matomo.org
+ * @license http://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
+ */
+namespace Piwik\Plugins\PagePerformance\Tracker;
+
+use Piwik\Common;
+use Piwik\Log;
+use Piwik\Plugin\Dimension\ActionDimension;
+use Piwik\Plugins\PagePerformance\Columns\TimeDomCompletion;
+use Piwik\Plugins\PagePerformance\Columns\TimeDomProcessing;
+use Piwik\Plugins\PagePerformance\Columns\TimeLatency;
+use Piwik\Plugins\PagePerformance\Columns\TimeOnLoad;
+use Piwik\Plugins\PagePerformance\Columns\TimeTransfer;
+use Piwik\Tracker;
+use Piwik\Tracker\Action;
+use Piwik\Tracker\ActionPageview;
+use Piwik\Tracker\Request;
+use Piwik\Tracker\RequestProcessor;
+use Piwik\Tracker\Visit\VisitProperties;
+
+/**
+ * Handles tracker requests containing performance data.
+ */
+class PerformanceDataProcessor extends RequestProcessor
+{
+    public function recordLogs(VisitProperties $visitProperties, Request $request)
+    {
+        /** @var null|Action $action */
+        $action = $request->getMetadata('Actions', 'action');
+
+        // update performance metrics only for non pageview requests
+        if ($action && $action instanceof ActionPageview) {
+            return;
+        }
+
+        $pageViewId = $request->getParam('pv_id');
+        $visitorId = $request->getVisitorId();
+
+        // ignore requests that can't be associated with an existing page view of a visitor
+        if (empty($pageViewId) || empty($visitorId)) {
+            return;
+        }
+
+        /** @var ActionDimension[] $performanceDimensions */
+        $performanceDimensions = [
```

Comment:
Be great to add some integration tests for below use case if possible. I suppose they might be already indirectly tested though through system tests. Maybe could just extract things into methods and test these methods.

eg one method fror line `48:68` -> `getPerformanceValuesToUpdate($request)` and `updatePerformanceValues()`

Reply:
will add some simple integration tests to check if updating those values in ongoing requests work as expected